src/Microsoft.DocAsCode.AzureMarkdownRewriters/AzureMigrationEngineBuilder.cs

```diff
@@ -266,5 +269,31 @@ private string GenerateAzureLinkHref(IMarkdownContext context, string href, stri
 
             return azureHref;
         }
+
+        private string GenerateAzureVideoLink(IMarkdownContext context, string azureVideoId, string rawMarkdown)
+        {
+            object path;
+            if (!context.Variables.TryGetValue("path", out path))
+            {
+                Logger.LogWarning("Can't get current file path. Skip video token rewriter.");
+                return azureVideoId;
+            }
+
+            if (!context.Variables.ContainsKey("azureVideoInfoMapping"))
+            {
+                Logger.LogWarning($"Can't fild the whole azure video info mapping. Current processing file: {path}, Raw: {rawMarkdown}");
+                return azureVideoId;
+            }
+
+            var azureVideoInfoMapping = (IReadOnlyDictionary<string, AzureVideoInfo>)context.Variables["azureVideoInfoMapping"];
+            if (azureVideoInfoMapping == null || !azureVideoInfoMapping.ContainsKey(azureVideoId))
```

Comment:
As above.

---
In reply to: [73269647](https://github.com/dotnet/docfx/pull/559#discussion_r73269647) [](ancestors = 73269647)